File: baekjoon/문자열/1157.py
word = input().lower()
alphabets = [
  'a',
  'b',
  'c',
  'd',
  'e',
  'f',
  'g',
  'h',
  'i',
  'j',
  'k',
  'l',
  'm',
  'n',
  'o',
  'p',
  'q',
  'r',
  's',
  't',
  'u',
  'v',
  'w',
  'x',
  'y',
  'z',
]
counts = []
for iIdex, i in enumerate(alphabets):
  counts.append(0)
for iIdx, i in enumerate(word):
  for jIdx, j in enumerate(alphabets):
    if i == j:
      counts[jIdx] = counts[jIdx] + 1
max = max(counts)
maxone = 0
maxtwo = 0
result = 0
try:
  maxone = counts.index(max)
  maxtwo = counts.index(max,maxone+1)
except:
  result = alphabets[maxone].upper()
else:
  result = "?"
print(result)

File: baekjoon/단계별로 풀어보기/문자열/1157.js
const readline = require('readline')
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

let input = []
const alphabets = [
  'a',
  'b',
  'c',
  'd',
  'e',
  'f',
  'g',
  'h',
  'i',
  'j',
  'k',
  'l',
  'm',
  'n',
  'o',
  'p',
  'q',
  'r',
  's',
  't',
  'u',
  'v',
  'w',
  'x',
  'y',
  'z',
]
let counts = []
for (i = 0; i < alphabets.length; i++) {
  counts[i] = 0
}

rl.on('line', (line) => {
  input.push(line)
}).on('close', () => {
  const word = input[0].toLowerCase()
  for (j = 0; j < word.length; j++) {
    for (i = 0; i < alphabets.length; i++) {
      if (word[j] === alphabets[i]) {
        counts[i]++
      }
    }
  }
  const max = Math.max.apply(null, counts)
  const result = counts.indexOf(max)
  const result2 = counts.indexOf(max, result + 1)
  if (result2 === -1) {
    console.log(alphabets[result].toUpperCase())
  } else {
    console.log('?')
  }
  process.exit()
})
